app/models.py

```diff
@@ -2142,3 +2148,56 @@ def serialize(self):
             "created_at": created_at_in_bst.strftime("%Y-%m-%d %H:%M:%S"),
         }
         return contact_list
+
+
+class BroadcastMessage(db.Model):
+    __tablename__ = 'broadcast_message'
+    __table_args__ = (
+        db.ForeignKeyConstraint(
+            ['template_id', 'template_version'],
+            ['templates_history.id', 'templates_history.version'],
+        ),
+        {}
+    )
+
+    STATES = [
+        'draft',
+        'pending-approval',
+        'rejected',  # rejected before leaving notify
+        'broadcasting',  # approved, now live
+        'completed',  # reached end_date
+        'cancelled',  # the broadcast went out but then we sent a second cancel message to end it prematurely
```

Comment:
A user needs to cancel a broadcast, does that need approval? Is there a state between requesting to cancel a broadcast and networks accepting that that request.

Reply:
don't think cancellation needs approval, but i'll double check with the big P.

no state between requesting cancellation and networks accepting. we set `cancelled` when the user clicks the button, then trigger the tasks. this state field isn't the place for figuring out whether the network received those messages (we'd need to have a new table with a row per provider at least)